Common/Data/Custom/USTreasury/YieldCurve.cs

```diff
@@ -0,0 +1,174 @@
+
+        /// <summary>
+        /// Five year yield curve
+        /// </summary>
+        public decimal? FiveYear { get; private set; }
+
+        /// <summary>
+        /// Seven year yield curve
+        /// </summary>
+        public decimal? SevenYear { get; private set; }
+
+        /// <summary>
+        /// Ten year yield curve
+        /// </summary>
+        public decimal? TenYear { get; private set; }
+
+        /// <summary>
+        /// Twenty year yield curve
+        /// </summary>
+        public decimal? TwentyYear { get; private set; }
+
+        /// <summary>
+        /// Thirty year yield curve
+        /// </summary>
+        public decimal? ThirtyYear { get; private set; }
+
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "alternative",
+                    "ustreasury",
+                    "yieldcurverates.csv"
+                ),
+                SubscriptionTransportMedium.LocalFile,
+                FileFormat.Csv
+            );
+        }
+
+        /// <summary>
+        /// Reads and parses yield curve data from a csv file
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="line">CSV line containing yield curve data</param>
+        /// <param name="date">Date request was made for</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>YieldCurve instance</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            // Date[0], 1 mo[1], 2 mo[2], 3 mo[3], 6 mo[4], 1 yr[5], 2 yr[6] 3 yr[7], 5 yr[8], 7 yr [9], 10 yr[10], 20 yr[11], 30 yr[12]
+            var csv = line.Split(new[] { ',' }, StringSplitOptions.None);
+            var csvDecimals = csv.Skip(1)
+                .Select(x => string.IsNullOrEmpty(x) ? (decimal?) null : Convert.ToDecimal(x, CultureInfo.InvariantCulture))
+                .ToList();
+
+            DateTime csvDate;
+            if (!DateTime.TryParseExact(csv[0], DateFormat.EightCharacter, CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal, out csvDate))
+            {
+                return null;
+            }
+
+            return new YieldCurve
+            {
+                // "These market yields are calculated from composites of indicative, bid-side
+                // market quotations (not actual transactions) obtained by the
+                // Federal Reserve Bank of New York at or near 3:30 PM each trading day"
+                //
+                // Remarks: Publication time was about an hour delayed on 2019-07-24 - verified by manual observation
+                // 20:30 UTC == 16:30 ET
+                Time = csvDate.Date.AddHours(16).AddMinutes(30),
```

Comment:
I think this is reasonable, assuming it's been verified on more than one occassion.